packages/aws-cdk-lib/aws-ec2/lib/vpc.ts

```diff
@@ -1491,7 +1491,7 @@ export class Vpc extends VpcBase {
 
     const createInternetGateway = props.createInternetGateway ?? true;
     const allowOutbound = this.subnetConfiguration.filter(
-      subnet => (subnet.subnetType !== SubnetType.PRIVATE_ISOLATED && subnet.subnetType !== SubnetType.ISOLATED)).length > 0;
+      subnet => (subnet.subnetType !== SubnetType.PRIVATE_ISOLATED && subnet.subnetType !== SubnetType.ISOLATED && !subnet.reserved)).length > 0;
```

Comment:
I honestly thought the following would be good here, but I'll leave it as is. (Because if there is no `PUBLIC` but `PRIVATE_WITH_EGRESS` in the code that follows, there will be an error.)

```suggestion
      subnet => (subnet.subnetType === SubnetType.PUBLIC && !subnet.reserved)).length > 0;
```

Reply:
The two implementations are different and `subnet => (subnet.subnetType === SubnetType.PUBLIC && !subnet.reserved)).length > 0;` would not allow the creation of internet or NAT gateways for `PRIVATE_WITH_EGRESS` subnets (unless at least a public subnet is present).
Can you please elaborate on why you considered making the change?

Reply:
> subnet => (subnet.subnetType === SubnetType.PUBLIC && !subnet.reserved)).length > 0; would not allow the creation of internet or NAT gateways for PRIVATE_WITH_EGRESS subnets (unless at least a public subnet is present).

The reason is that trying to create Nat gateways with `PRIVATE_WITH_EGRESS` without a public subnet in the first place would result in an error.

https://github.com/aws/aws-cdk/blob/v2.118.0/packages/aws-cdk-lib/aws-ec2/lib/vpc.ts#L2416-L2419

As for the Internet Gateway, I questioned whether there was a need to create it without a public subnet (like the following code). However, since it is not good to carelessly change the behavior, I left it as it was, just in case.

```ts
      new Vpc(stack, 'TheVPC', {
        subnetConfiguration: [
          {
            subnetType: SubnetType.PRIVATE_WITH_EGRESS,
            name: 'egress',
          },
        ],
        natGateways: 0,
      });
```

Reply:
Thanks for the clarification.
Yeah, I'd just leave it as is unless we have a specific reason to change it to avoid potential errors with existing deployments.